client/uploadmanager.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * @license Apache-2.0
+ *
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//goog.provide('filedrop.UploadManager');
+var filedrop = filedrop || {};
+
+
+/**
+ * Service for managing uploads.
+ * @param {!angular.$q} $q
+ * @param {!filedrop.Files} Files
+ * @constructor
+ * @ngInject
+ */
+filedrop.UploadManager = function($q, Files, $log) {
+  /** @private {!angular.$q} */
+  this.q_ = $q;
+
+  /** @private {!filedrop.Files} */
+  this.filesService_ = Files;
+
+  /** @private {!angular.$log} */
+  this.log_ = $log;
+
+  /** @export {!Object} */
+  this.uploads = {};
+};
+
+/**
+ * Adds promise to central upload store.
+ * @param {!File} file
+ * @return {string} upload object id
+ */
+filedrop.UploadManager.prototype.add = function(file) {
+  var upload = this.filesService_.upload(file);
+  var id = file.name;
```

Comment:
Make this an incrementing counter instead so as to avoid overwriting existing uploads.